app/controllers/needs_controller.rb

```diff
@@ -20,8 +20,28 @@ def index
     else
       @needs = @needs.order(created_at: :desc)
     end
+    page_size = params[:page_size].to_i > 0 ? params[:page_size] : 10
+    @needs = @needs.page(params[:page]).per(page_size)
 
-    @needs = @needs.page(params[:page])
+  end
+
+  def export
+    @params = params.permit(:user_id, :status, :category, :page, :order_dir, :order, :commit)
```

Comment:
This section should include the `:is_urgent` bits from my PR if we want to allow filtering/sorting by that param too.